Expose PrimesElem as #math.primes in Typst code

Leedehai opened the pull request:
Hi - when writing packages I found sometimes I could not do `content.func() == math.foo-bar` before introspecting fields in  `FooBarElem`. This PR fixes this issue and adds a test.

In particular, the list added three missing elements: `MathStyleElem` (which motivated this PR), `AlignPointElem`, and `PrimesElem`.

laurmaedje commented on tests/typ/math/math-module.typ:
```diff
@@ -0,0 +1,45 @@
+// Tests the "math" module has the elems
```
This test is unnecessary imo, since it exactly the mirror the list in mod.rs. It doesn't add any new information.

laurmaedje commented on crates/typst/src/math/mod.rs:
```diff
@@ -158,37 +158,60 @@ pub static MATH: Category;
 pub fn module() -> Module {
     let mut math = Scope::deduplicating();
     math.category(MATH);
-    math.define_elem::<EquationElem>();
```
I'd prefer to keep the elements listed as before. The file comments just add duplication with the real file system layout (that could get out of date) and also introduce inconsistency with the other categories (non-math).

laurmaedje reviewed (commented):
- MathStyleElem is supposed to be an implementation detail (unfortunately it can't be because there is no privacy concept). Adding it to the scope like this would also generate a docs page for it, but I don't want to expose it in this way. It is supposed to be removed in the future in favor of using the normal styling mechanisms.

- AlignPointElem wasn't exposed because I don't like the name. But I guess we can expose it (or perhaps find a better name?).

- PrimesElem can be exposed I think. That's a latter addition that has just been forgotten.

If we expose an element, we need to ensure that it has at least a bare minimum of documentation.

Leedehai commented:
Hi - since MathStyleElem is an implementation detail and therefore should be hidden from users, it probably shouldn't be an Elem at all? In the status quo, there's no good way to inspect the body of a MathStyleElem because the code can't check `content.func() == math.math-style ` first. This difficulty makes MathStyleElem _not really_ "hidden".

For example, I'd like to write this. Without this PR, it will return different results for inputs `a+b/2` and `display(a+b/2)`, but users will expect the same result because `display(..)` just displays things a bit differently.

```typ
#let __is_add_sub_sequence(content) = {
  if not content.has("children") { return false }

  let impl(seq) = {
    // Only check the top level, don't descend into LR groups
    for child in seq.at("children") {
      if child == [+] or child == [#sym.minus] { return true }
    }
    return false
  }

  // Before https://github.com/typst/typst/pull/3063:
  return impl(content)
  // After https://github.com/typst/typst/pull/3063:
  // if content.func() == math.math-style {
  //   return impl(content.at("body"))
  // } else {
  //   return impl(content)
  // }
}
```

laurmaedje commented:
> since MathStyleElem is an implementation detail and therefore should be hidden from users, it probably shouldn't be an Elem at all?

As said above, it is supposed to be removed in the future in favor of using the normal styling mechanisms. It is a hack.

> This difficulty makes MathStyleElem not really "hidden": it has observable effects in coding.

That's what I mean with "unfortunately it can't be because there is no privacy concept".

In the future, `display` should result in a `styled` element, just like a normal set rule does.

Leedehai commented:
Thanks! So this PR only adds the forgotten `PrimesElem`, and I modified the PR's title accordingly.

As for `AlignPointElem`: removed from the PR since you have not decided on a better name yet :) it's not that "urgent" to expose it - I guess there's not many use cases for it anyway. But long term, I believe it's better to expose it.

As for `MathStyleElem`: removed from the PR, since that elem will be refactored as mentioned above.

laurmaedje commented:
Great, thanks!
